### PR TITLE
HTTP regression tests

### PR DESCRIPTION
**Pull request by markrtuttle**
This pull request adds a method of doing regression tests using the FreeRTOS HTTP proofs.

Review this pull request commit by commit for a natural grouping of the files being changed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/build-viewer-reports [hunk 1]
```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+
+# Set emacs mode to Python
+# -*- mode: Python;-*-
```
Comment (feliperodri) [inline tests/build-viewer-reports:4]: Why is this necessary? Probably to make `emacs` users happy, which is fine 😄

File: tests/Makefile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+SCRIPTS = \
+  build-viewer-reports \
+  compare-result \
+  compare-source \
+  compare-viewer-reports
+
+default:
+	@ echo Nothing to do for $@
```
Comment (feliperodri) [inline tests/Makefile:8]: We should add a comment explaining why `default` is set to nothing. Something like "makes sure users are ware which operation is performed by this Makefile".

File: tests/build-viewer-reports [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+             for ci_stage in pipeline['ci_stages']
+             for job in ci_stage['jobs']
+             if job['wrapper_arguments']['ci_stage'] == 'report']
+    return [{
+        COMMAND: job['wrapper_arguments']['command'],
+        REPORT: job['wrapper_arguments']['outputs'][0]
+    } for job in jobs]
+
+################################################################
+# Re-generate the cbmc-viewer reports
```
Comment (feliperodri) [inline tests/build-viewer-reports:146]: It'd be nice to add here "why do you need to re-generate" the reports.

File: tests/compare-result [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          # Ignore reported version number for CBMC
+          not line.startswith('CBMC version') and
+          # Ignore reported runtimes
+          not line.startswith('Runtime') and
+          # Ignore path to goto binary
+          not line.startswith('Reading') and
+          # Ignore status lines with embedded path names
+          not line.startswith('Unwinding loop') and
+          not line.startswith('aborting path') and
+          not line.startswith('Not unwinding loop')
```
Comment (feliperodri) [inline tests/compare-result:48]: Are there any other cases that could include path names? If yes, what would it be the impact for this script if we miss it?

File: tests/compare-viewer-reports [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+RESULT_JSON = 'viewer-result.json'
+SOURCE_JSON = 'viewer-source.json'
+OTHER_JSON = [
+    'viewer-coverage.json',
+    'viewer-loop.json',
+    'viewer-property.json',
+    'viewer-reachable.json',
+    'viewer-symbol.json',
+    'viewer-trace.json'
+]
```
Comment (feliperodri) [inline tests/compare-viewer-reports:23]: What if this list grows? Do we warn the user if we find `json` files not listed in here?

File: tests/compare-viewer-reports [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    source_path = json_path/SOURCE_JSON
+    logging.info("Comparing %s", source_path)
+    equal_json = compare_source(reports1/source_path, reports2/source_path) and equal_json
+
+    for other in OTHER_JSON:
+        other_path = json_path/other
+        logging.info("Comparing %s", other_path)
+        equal_json = compare_diff(reports1/other_path, reports2/other_path, verbose) and equal_json
+
+    return equal_json
```
Comment (feliperodri) [inline tests/compare-viewer-reports:199]: In the future, we might want to use [difflib](https://docs.python.org/3/library/difflib.html) or [json-diff](https://pypi.org/project/json-diff/) instead of textual diffs. Shall we at least create a GitHub issue now so we can track this improvements in the future?

File: tests/coreHTTP/.gitignore [hunk 7]
```diff
@@ -0,0 +1,3 @@
+coreHTTP/
+reports1/
+reports2/
```
Comment (feliperodri) [inline tests/coreHTTP/.gitignore:3]: Add newline at the end of the file.

File: tests/coreHTTP/run-cbmc-proofs.patch [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+         sys.exit(1)
+-    return proc.stdout.strip()
+-
++    return strip_make_output(proc.stdout)
+ 
+ def get_litani_capabilities(litani_path):
+     cmd = [litani_path, "print-capabilities"]
+-- 
+2.30.1 (Apple Git-130)
+
```
Comment (feliperodri) [inline tests/coreHTTP/run-cbmc-proofs.patch:39]: We should open a GitHub issue at least to request this update on Litani. Why would we need an specific version of this script just for these regression tests?

File: tests/build-viewer-reports (hunk 1 above)
Comment (markrtuttle) [reply tests/build-viewer-reports:4]: It is important to make emacs users happy...

File: tests/compare-viewer-reports (hunk 5 above)
Comment (markrtuttle) [reply tests/compare-viewer-reports:23]: Added a warning.

File: tests/coreHTTP/.gitignore (hunk 7 above)
Comment (markrtuttle) [reply tests/coreHTTP/.gitignore:3]: done
